src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,353 @@
+  ...mockEventBubble,
+  relatedTarget: {
+    dataset: {
+      shiftFocus: 'last-opened-toolbar-trigger-button',
+    },
+  },
+};
+
+const renderVisualRefreshTriggerButton = (
+  props: Partial<VisualRefreshTriggerButtonProps> = {},
+  useAppLayoutInternalsValues: Partial<AllContext.AppLayoutInternals> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  MockUseAppLayoutInternals.mockReturnValue({
+    ...mockUseLayoutInternalValues,
+    ...useAppLayoutInternalsValues,
+  } as AllContext.AppLayoutInternals);
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshTriggerButton {...(renderProps as VisualRefreshTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(visualRefreshStyles['trigger-wrapper'])!;
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+const renderVisualRefreshToolbarTriggerButton = (
+  props: Partial<VisualRefreshToolbarTriggerButtonProps> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshToolbarTriggerButton {...(renderProps as VisualRefreshToolbarTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(toolbarTriggerButtonStyles['trigger-wrapper'])!;
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+describe('Visual refresh trigger-button (not in appLayoutWidget toolbar)', () => {
+  beforeEach(() => jest.clearAllMocks());
+
+  //tests for both desktop and mobile
+  describe.each([true, false])('Toolbar trigger-button with isMobile=%s', isMobile => {
+    //tests for both drawer being opend and closed
+    describe.each([true, false])('AppLayoutInternals with hasOpenDrawer=%s', hasOpenDrawer => {
+      test('renders correctly with without badge', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(wrapper.findByClassName(visualRefreshStyles.dot)).toBeNull();
+        expect(wrapper.findBadge()).toBeNull();
+      });
+
+      test('renders correctly with aria controls', () => {
```

Comment:
Can this test be merged with the previous one? It looks like it only verifies if the ariaControls property is assigned.

Reply:
If not merge, I would at least change the description to something like "aria controls is assigned on the button"